### ISSUE TITLE
Evaluation and Export

### ISSUE THREAD

**Issue by supreme-gg-gg** [gemma-facet/cloud-services#13]
- [x] Hugging face and unsloth merging model export, [unsloth docs](https://docs.unsloth.ai/basics/running-and-saving-model), [hf docs on this](https://huggingface.co/docs/peft/v0.17.0/en/package_reference/lora)
- [x] Evaluation during training (_does not work on both HF and Unsloth_, known to be blocked on unsloth, should work on HF but goes OOM when trying to eval)
- [x] Evaluation at inference service (using `evaluate`)
- [ ] Hugigng face to GGUF (this requires `llama.cpp`, will ship soon)
- [x] Unsloth GGUF

NOTE: unsloth eval during training is also blocked 💀, i think we just migrate the eval suite to inference and do it after training separately from `SFTTrainer` because currently unsloth makes it not to return logits...

**Comment by supreme-gg-gg** [comment]
More comprehensive evaluation can use "metric" from "evaluate": https://huggingface.co/docs/evaluate/en/transformers_integrations

**Comment by supreme-gg-gg** [comment]
Types for metrics:

- Generic metrics (we support this for now)
- Task specific metrics (this will come in another PR / Issue, we will support a few common ones for task specific adaptations!) -- required more research
- Dataset specific (not planned)

**Comment by supreme-gg-gg** [comment]
GGUF for hugging face transformers requires setting up llama.cpp first and run the convert script: https://huggingface.co/docs/transformers/en/gguf
